src/net_processing.cpp

```diff
@@ -3636,6 +3636,10 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     }
     CNetMessage& msg(msgs.front());
 
+    std::vector<unsigned char> tmp(msg.m_recv.data(), msg.m_recv.data() + msg.m_recv.size());
```

Comment:
The scope of `tmp` can be limited to the `gArgs.IsArgSet("-logmessages")` case below.

Reply:
good point, done